x-pack/legacy/plugins/ml/server/routes/job_validation.ts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { RequestHandlerContext } from 'src/core/server';
+import { schema } from '@kbn/config-schema';
+import { licensePreRoutingFactory } from '../new_platform/licence_check_pre_routing_factory';
+import { wrapError } from '../client/error_wrapper';
+import { RouteInitialization } from '../new_platform/plugin';
+import {
+  estimateBucketSpanSchema,
+  modelMemoryLimitSchema,
+  validateJobSchema,
+} from '../new_platform/job_validation_schema';
+import { anomalyDetectionJobSchema } from '../new_platform/anomaly_detectors_schema';
+import { estimateBucketSpanFactory } from '../models/bucket_span_estimator';
+import { calculateModelMemoryLimitProvider } from '../models/calculate_model_memory_limit';
+import { validateJob, validateCardinality } from '../models/job_validation';
+
+/**
+ * Routes for job validation
+ */
+export function jobValidationRoutes({ config, xpackMainPlugin, router }: RouteInitialization) {
+  function calculateModelMemoryLimit(context: RequestHandlerContext, payload: any) {
```

Comment:
NIT: the handler calling that seems to have a proper `body` validation schema. Could probably get rid of the `any` signature here
```ts
type youWillNameThatBetterThanMePayload = TypeOf<typeof schema.object(modelMemoryLimitSchema)>
```

Reply:
This I did not know about and it is beautiful. I thank you 🙏